### ISSUE TITLE
Query Builder unsets unrelated "queries" from JetSmartFilterSettings when query id is empty.

### ISSUE THREAD

**Issue by lukechinworth** [CrocoBlock/suggestions#5636]
**Describe the bug**
Using a listing grid on a category archive template, with several smart filters using "mixed". Then I have another listing grid on the same page listing out the other categories, which are pulled in using the query builder (this query does not have a query id set). If I select a filter, and then reload with the filtered url, the query builder module clears out the JetSmartFilterSetting["queries"] at jet-engine/includes/components/query-builder/listings/filters.php:206. This means that when I remove the filter (change the select back to "all"), the listing grid does not use the archive template query, but the settings on the listing grid, which are normally never applied.

**To Reproduce**
1. Add a category archive template with the theme builder. add a listing grid, and select "use as archive template".
2. Add a smart filter for another taxonomy, and select "mixed"
3. On the archive template, add another listing grid that uses a custom query from the query builder. On the the query builder settings, leave the query id empty.
4. Load the page and select a smart filter.
5. reload the page with the filtered url.
6. Remove the smart filter by changing the filter back to the placeholder "all"
7. BUG: the listing grid displays posts according to the listing grid settings - listing post type, post num.

**Expected behavior**
The listing grid should display posts according to the category archive template query.

**Desktop (please complete the following information):**
 - OS: macOS Catalina
 - Browser Chrome 103
 - JetEngine 3.0.1
 - JetSmartFIlters 2.3.12

**Comment by MjHead** [comment]
@lukechinworth Such case was on of the main reasons of adding Query ID option for the Queries. We just investigated this issue one more time and there is still the only solution.

Where is the problem:
- Query ID option was not added from the first version of Query Builder, so from the start query builder queries was separated only by listing ID and filter ID. 

But on some stage we undersatnd that for some cases we can't correctly define what query on the page should be filtered withou connecting exactly the Query to filter by ID, not just Listing Grid. On this stage was added Query ID option and it could seprate queries should be filtered for all cases.

But due to compatibility reasons we must keep an initial behavior - when only Listing ID and Filter ID is set, if we change this - for a lot people, who set up their filter unstil adding Query ID option, filterin will be broken.

So at the moment the only solution for your case - is set Query ID option for Query.

We constantly working on improwing this part of JetEngine/JSF, so if at some point we'll find the solution for such cases without breaking backward compatibility - we'll implement it.
